examples/demo.rs

```diff
@@ -0,0 +1,55 @@
+use std::io::{self, Read, prelude::*};
+use std::time::{SystemTime, UNIX_EPOCH, Duration};
+use std::net::TcpStream;
+use std::thread;
+use std::str;
+
+
+
+fn main() -> io::Result<()> {
+    let mut stream = TcpStream::connect("localhost:5555").expect("couldn't connect to server");
+    let mut stream_clone = stream.try_clone().expect("clone failed...");
+    println!("Connected to the server!");
+ 
+    // Thread for sending data
+    thread::spawn(move || {
+        send_device_shadow(stream_clone);
+    });
+    
+    // receives and prints json data
+    let mut line = [0;2048];
+    loop{
+       let result = stream.read(&mut line)?;
+       let data = str::from_utf8(&line[0..result]).unwrap();
```

Comment:
```suggestion
    for data in stream.lines() {
```
This would also significantly simplify the code. And you need to only import `std::io::BufRead`